File: public/SwrveSDK/SDK/Messaging/SwrveStoryDismissButton.m
#import "SwrveStoryDismissButton.h"

@implementation SwrveStoryDismissButton

@synthesize buttonId;
@synthesize name;
@synthesize color;
@synthesize pressedColor;
@synthesize focusedColor;
@synthesize size;
@synthesize marginTop;
@synthesize accessibilityText;

- (instancetype)initWithDictionary:(NSDictionary *)storySettings {
    self = [super init];
    if (self) {
        self.buttonId = [storySettings objectForKey:@"id"];
        self.name = [storySettings objectForKey:@"name"];
        self.color = [storySettings objectForKey:@"color"];
        self.pressedColor = [storySettings objectForKey:@"pressed_color"];
        self.focusedColor = [storySettings objectForKey:@"focused_color"];
        self.size = [storySettings objectForKey:@"size"];
        self.marginTop = [storySettings objectForKey:@"margin_top"];
        self.accessibilityText = [storySettings objectForKey:@"accessibility_text"];
    }
    return self;
}

@end

File: SwrveSDKSwift/SwrveStoryDismissButton.swift
import UIKit

@objc public class SwrveStoryDismissButton: NSObject {
    @objc public var buttonId: NSNumber?
    @objc public var name: String?
    @objc public var color: String?
    @objc public var pressedColor: String?
    @objc public var focusedColor: String?
    @objc public var size: NSNumber?
    @objc public var marginTop: NSNumber?
    @objc public var accessibilityText: String?

    @objc(initWithDictionary:)
    public init(storySettings: [String: Any]) {
        self.buttonId = storySettings["id"] as? NSNumber
        self.name = storySettings["name"] as? String
        self.color = storySettings["color"] as? String
        self.pressedColor = storySettings["pressed_color"] as? String
        self.focusedColor = storySettings["focused_color"] as? String
        self.size = storySettings["size"] as? NSNumber
        self.marginTop = storySettings["margin_top"] as? NSNumber
        self.accessibilityText = storySettings["accessibility_text"] as? String
        super.init()
    }
}
